docker/Dockerfile

```diff
@@ -1,7 +1,21 @@
+# Stage -1: Update apt-get
+FROM ubuntu:jammy AS base-ubuntu
+
+# Mount apt's cache folders to cache install but mainstain
+# reproducibility on local machine. More details at:
+# https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---mounttypecache
+
+# Set sharing=locked to avoid race condition with between stages.
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update && apt-get upgrade -y
```

Comment:
Not sure if this would work, as ubuntu runs [`apt-get clean` after `apt-get` invocations](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get). Let's not do this, and keep to the best practices.

Reply:
I came across (this type of caching)[https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---mounttypecache] while scrolling through Moby's docs, and I guess it should be working. I have also added apt-get update commands before other apt-get calls to keep to the best practices.

If this works, then local build is sped up significantly. As for production builds, the workflow files currently do not enable caching since apt behavior is not really reproducible (as seen with python3.6) and probably won't, so I guess there's no change.